sdk/java/src/main/java/com/chain/http/Client.java

```diff
@@ -633,28 +652,89 @@ public Builder setAccessToken(String accessToken) {
     }
 
     /**
+     * Sets the client's certificate and key for TLS client authentication.
+     * @param certStream input stream of pem encoded X.509 certificate
+     * @param keyStream input stream of pem encoded private key
+     */
+    public Builder setX509KeyPair(InputStream certStream, InputStream keyStream)
+        throws HTTPException {
+      try (PEMParser parser = new PEMParser(new InputStreamReader(keyStream))) {
+        // Extract certs from PEM-encoded input.
+        CertificateFactory factory = CertificateFactory.getInstance("X.509");
+        X509Certificate certificate = (X509Certificate) factory.generateCertificate(certStream);
+
+        // Parse the private key from PEM-encoded input.
+        Object obj = parser.readObject();
+        PrivateKeyInfo info;
+        if (obj instanceof PEMKeyPair) {
+          // PKCS#1 Private Key found.
+          PEMKeyPair kp = (PEMKeyPair) obj;
+          info = kp.getPrivateKeyInfo();
+        } else if (obj instanceof PrivateKeyInfo) {
+          // PKCS#8 Private Key found.
+          info = (PrivateKeyInfo) obj;
+        } else {
+          throw new HTTPException("Unsupported private key provided.");
+        }
+
+        // Create a new key store and input the pair.
+        KeySpec spec = new PKCS8EncodedKeySpec(info.getEncoded());
+        KeyFactory kf = KeyFactory.getInstance("RSA");
+        KeyStore keyStore = KeyStore.getInstance(KeyStore.getDefaultType());
+        keyStore.load(null, DEFAULT_KEYSTORE_PASSWORD);
+        keyStore.setCertificateEntry("cert", certificate);
+        keyStore.setKeyEntry(
+            "key",
+            kf.generatePrivate(spec),
+            DEFAULT_KEYSTORE_PASSWORD,
+            new X509Certificate[] {certificate});
+
+        // Use key store to build a key manager.
+        KeyManagerFactory keyManagerFactory =
+            KeyManagerFactory.getInstance(KeyManagerFactory.getDefaultAlgorithm());
+        keyManagerFactory.init(keyStore, DEFAULT_KEYSTORE_PASSWORD);
+
+        this.keyManagers = keyManagerFactory.getKeyManagers();
+        return this;
+      } catch (GeneralSecurityException | IOException ex) {
+        throw new HTTPException("Unable to store X.509 cert/key pair", ex);
+      }
+    }
+
+    /**
+     * Sets the client's certificate and key for TLS client authentication.
+     * @param certPath file path to pem encoded X.509 certificate
+     * @param keyPath file path to pem encoded private key
+     */
+    public Builder setX509KeyPair(String certPath, String keyPath) throws HTTPException {
+      try (InputStream certStream =
+              new ByteArrayInputStream(Files.readAllBytes(Paths.get(certPath)));
+          InputStream keyStream =
+              new ByteArrayInputStream(Files.readAllBytes(Paths.get(keyPath)))) {
+        return setX509KeyPair(certStream, keyStream);
+      } catch (IOException ex) {
+        throw new HTTPException("Unable to store X509 cert/key pair", ex);
```

Comment:
I think HTTPException is a weird error to throw here. HTTPException is a transport-level error during a request. We may need to make a new error class that represents configuration errors.

Might want to double-check that HTTPException is being correctly used in other sites as well.